pkg/ibmcsidriver/controller.go

```diff
@@ -140,8 +140,88 @@ func (csiCS *CSIControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
+		VolumeID:          volumeObj.VolumeID,
+		VPCID:             os.Getenv("VPC_ID"),
+		AccessControlMode: requestedVolume.AccessControlMode,
+	}
+
+	/*
+		//IF ENI/VNI is enabled
+
+		Case 1: User has not provided anything.
+		The VolumeAccessPoint (aka File share target) will be created with mountPath having randomIP from subnet within the same zone as used by the
+		volume (aka File share). The zone is picked-up randomly from topology and the subnet is fetched via the CSI driver matching with the
+		volumeaccess point zone and cluster subnet list.In this case any random IP Address will be created and assigned to VNI in the fetched subnet range.
+
+		Case 2: User has provided the subnetId, zone but nothing else
+		The VolumeAccessPoint (aka File share target) will be created with mountPath having randomIP from subnet within the zone provided by user
+		as used by the volume (aka File share).In this case any random IP Address will be created and assigned to VNI in the user provided subnet range.
+
+		Case 3: User has provided the subnetId, zone and PrimaryIPAdrees
+		The VolumeAccessPoint (aka File share target) will be created with mountPath having PrimaryIPAdrees from subnet within the zone provided by user
+		as the volume (aka File share).In this case any PrimaryIPAdrees will be created and assigned to VNI in the user provided subnet range.
+
+		Case 4: User has provided the subnetId, zone and PrimaryIPID
+		The VolumeAccessPoint (aka File share target) will be created with mountPath having IP adress associated with PrimaryIPID from subnet within
+		the zone provided by user as used by the volume (aka File share).In this case any IP adress associated with PrimaryIPID is assigned to VNI.
+
+		Case 5: User has not provided the subnetId and provided zone, PrimaryIPID
+		The VolumeAccessPoint (aka File share target) will be created with mountPath having IP adress associated with PrimaryIPID from subnet within
+		the zone provided by user as used by the volume (aka File share).In this case any IP adress associated with PrimaryIPID is assigned to VNI.
+
+		Case 6: User has not provided the subnetId,zone but provided PrimaryIPID
+		This will throw error that zone is mandatory as CSI driver cannot predict the zone in such scenarios CSI will pick this up from topology.
+
+		Case 7: User has provided the subnetId but nothing else
+		This will throw error that zone is mandatory as CSI driver cannot predict the zone in such scenarios CSI will pick this up from topology.
+
+		Case 8: User has not provided the subnetID but provided the zone  and PrimaryIPAddress
+		This will throw error that subnet is mandatory as CSI driver cannot predict the subnet in such scenarios as there
+		might be multiple subnets in same zone.
+
+		Case 9: User has provided the subnetID and PrimaryIPAddres but not provided the zone.
+		This will throw error that zone is mandatory as CSI driver cannot predict the zone in such scenarios CSI will pick this up from topology.
```

Comment:
If user has given subnetID we need to fetch zone before file share creation. We can consider this in next PR until that lets have zone has mandatory in such scenarios